snmp-scan.py

```diff
@@ -121,6 +121,18 @@ def scan_host(ip):
         except (herror, gaierror):
             pass
 
+        # If there is no hostname and the configuration option to add by IP is
+        # not set, we should not add this to LibreNMS
+        if hostname is None and not CONFIG.get('discovery_by_ip'):
+            try:
```

Comment:
Why are you trying to ping it?
Also, you can't really hardcode the path of fping, it is not the same on many OS.

Reply:
Thanks for reviewing @murrant. Sure, no problem. `fping` can be abstracted. If you have `discovery_by_ip` set to `False` so you don't want to discover by IP the only output we can get back is `FAILED` - the ping test at least gets us the `UNPINGABLE` result if appropriate too.

Reply:
Yes, but it will check things below, so I don't see why you would need to ping it here anyway...

Reply:
Maybe the result should be "SKIPPED" no need to ping...

Reply:
`UNPINGABLE` provides more information than `EXCLUDED`, but as we're here talking about functionality where we don't want to `addhost` when there's no reverse DNS in place then it is reasonable to simply return `EXCLUDED` - I'll make that change.